addons/l10n_sa/models/account_move.py

```diff
@@ -50,9 +49,9 @@ def get_qr_encoding(tag, field):
     def _post(self, soft=True):
         res = super()._post(soft)
         for record in self:
-            if record.country_code == 'SA' and record.move_type in ('out_invoice', 'out_refund'):
-                if not record.l10n_sa_show_delivery_date:
-                    raise UserError(_('Delivery Date cannot be empty'))
+            if record.country_code == 'SA' and record.is_sale_document():
+                if not record.delivery_date:
+                    record.delivery_date = record.invoice_date
                 self.write({
                     'l10n_sa_confirmation_datetime': fields.Datetime.now()
                 })
```

Comment:
```suggestion
            if record.country_code == 'SA' and record.is_sale_document():
                vals = {'l10n_sa_confirmation_datetime': fields.Datetime.now()}
                if not record.delivery_date:
                    vals['delivery_date'] = record.invoice_date
                record.write(vals)
```
I just think it's weird to use 2 methods for doing the same thing (setting a field value on a move) at the same place :sweat_smile: 
It would be nice to use the same (either one).
If I'm not mistaken, the `write` should be done on the record instead of self.
And if you feel like it you can also change the (annoying/generic/one-size-fits-all) `record` variable name to `move` :innocent: